2025.09 NEWS Grammar Updates (#16406)
rstudio/rstudio: version/news/NEWS-2025.09.0-cucumberleaf-sunflower.md

#### RStudio

- ([#16293](https://github.com/rstudio/rstudio/issues/16293)): Added a Source column that displays a package's origin (if known) in RStudio’s Packages pane
- ([#16294](https://github.com/rstudio/rstudio/issues/16294)): Provided an additional icon in RStudio’s Packages pane that links to a package’s external documentation (e.g., pkgdown website) when available

> - ([#16295](https://github.com/rstudio/rstudio/issues/16295)): Provide UI in RStudio for changing the current repository/snapshot when using Posit Package Manager

Reviewer: This might read a little easier?

```suggestion
- ([#16295](https://github.com/rstudio/rstudio/issues/16295)): Added functionality in the RStudio UI for changing the current repository/snapshot when using Posit Package Manager
```

- ([#15841](https://github.com/rstudio/rstudio/issues/15841)): Added support for highlighing all SQL 2023 standard keywords in SQL documents in RStudio
- ([#16138](https://github.com/rstudio/rstudio/issues/16138)): Added support for version 17 of the R graphics engine in RStudio
- ([#16213](https://github.com/rstudio/rstudio/issues/16213)): Improved legibility of highlighted code when the RStudio debugger is active
- ([#15945](https://github.com/rstudio/rstudio/issues/15945)): Added a user preference in RStudio to disable showing the splash screen at startup
- ([#15614](https://github.com/rstudio/rstudio/issues/15614)): Added support for closing the RStudio splash screen when clicked with the mouse
- ([#16111](https://github.com/rstudio/rstudio/issues/16111)): Increased RStudio’s default console buffer size to 10,000 lines

> - ([#11661](https://github.com/rstudio/rstudio/issues/11661)): Bring an existing zoomed plot window to the foreground when clicking the Zoom button in the RStudio Desktop plots pane

Reviewer: ```suggestion
- ([#11661](https://github.com/rstudio/rstudio/issues/11661)): Added functionality to bring an existing zoomed plot window to the foreground when clicking the Zoom button in the RStudio Desktop plots pane
```

- ([#16214](https://github.com/rstudio/rstudio/issues/16214)): Introduced a new structure for RStudio release note entries
- ([#16226](https://github.com/rstudio/rstudio/issues/16226)): Added an option (off by default) to zoom RStudio Desktop UI with Ctrl/Cmd + mouse wheel

#### Posit Workbench

- ([#14083](https://github.com/rstudio/rstudio/issues/14083)): Added support for custom certificate bundles for GitHub Copilot in RStudio
- (rstudio-pro#8093): Redesigned the Workbench homepage to provide a more efficient and intuitive user experience focused on projects
- (rstudio-pro#6768): Added Positron and VS Code projects alongside RStudio projects on the Workbench homepage
- (rstudio-pro#8162): Added a custom “Metadata” column in the RStudio Packages pane for sessions using Posit Package Manager
- (rstudio-pro#5357): Added support for enforcing the user limit specified by the Posit Workbench product license strictly
- (rstudio-pro#7533): Released Posit Product Documentation theme v7.0.1 with cookie consent, style updates, accessibility fixes, and dark theme improvements
- (rstudio-pro#8186): Enabled Positron Pro sessions by default in Workbench

> - (rstudio-pro#7599): Added a step to back-up of existing Positron Pro configuration files and install defaults during upgrade

Reviewer: ```suggestion
- (rstudio-pro#7599): Added a step to back-up existing Positron Pro configuration files and install defaults during upgrade
```

- (rstudio-pro#8072): Added support for importing VS Code settings into Positron Pro in Workbench
- (rstudio-pro#7466): Added support for setting the default Connect server for the Publisher extension in Positron settings via the `default-rsconnect-server` option in `rsession.conf`
- (rstudio-pro#8697): Added a product check request in Workbench that notifies a Posit server when the service is set up and actively using a license. Control this with the `server-product-check` configuration setting (enabled by default: `1`)`.
- (rstudio-pro#8090): Improved integration between RStudio and Posit Package Manager, including repository discovery, authentication, and feature visibility
- (rstudio-pro#5068): Added support for just-in-time provisioning of local groups for Proxied Auth, OpenID Connect, and SAML Authentication in Workbench
- (rstudio-pro#7817): Added a Run Script in a Workbench Job action (and matching command) in the editor toolbar for R and Python scripts in Positron and VS Code

> - (rstudio-pro#8463): Organized the Workbench Jobs pane in Positron and VS Code by script, adopt familiar VS Code UI conventions, and open scripts or virtual Job Details documents when clicked

Reviewer: ```suggestion
- (rstudio-pro#8463): Organized the Workbench Jobs pane in Positron Pro and VS Code by script, adopt familiar VS Code UI conventions, and open scripts or virtual Job Details documents when clicked
```

> - (rstudio-pro#7862): Added support for dismissing completed jobs individually from the Workbench Jobs pane in Positron and VS Code. Provided Dismiss Completed Jobs and Restore Completed Jobs actions to hide/unhide all completed jobs (not persistent across sessions).

Reviewer: ```suggestion
- (rstudio-pro#7862): Added support for dismissing completed jobs individually from the Workbench Jobs pane in Positron Pro and VS Code. Provided Dismiss Completed Jobs and Restore Completed Jobs actions to hide/unhide all completed jobs (not persistent across sessions).
```

> - (rstudio-pro#8464): Added a preview setting in Workbench to launch jobs from Positron and VS Code with managed credentials (disabled by default)

Reviewer: ```suggestion
- (rstudio-pro#8464): Added a preview setting in Workbench to launch jobs from Positron Pro and VS Code with managed credentials (disabled by default)
```

- (rstudio-pro#8148): Provided a new Audit Database and `get_historical_session` Workbench API endpoint to retrieve data about ended sessions

> - (rstudio-pro#8124): Made Prometheus metrics generally available and remove Preview status

Reviewer: ```suggestion
- (rstudio-pro#8124): Made Prometheus metrics generally available and removed Preview status
```

> - (rstudio-pro#8289): Removed `pwb_session_duration`from Prometheus Metrics and provide session start/end times via the new Workbench API endpoint `get_historical_session`

Reviewer: ```suggestion
- (rstudio-pro#8289): Removed `pwb_session_duration`from Prometheus Metrics and provided session start/end times via the new Workbench API endpoint `get_historical_session`
```

- (rstudio-pro#8090): Improve integration between RStudio and Posit Package Manager, including repository discovery, authentication, and feature visibility
- (rstudio-pro#5068): Support just-in-time provisioning of local groups for Proxied Auth, OpenID Connect, and SAML Authentication in Workbench
- (rstudio-pro#7817): Add a Run Script in a Workbench Job action (and matching command) in the editor toolbar for R and Python scripts in Positron and VS Code
- (rstudio-pro#8463): Organize the Workbench Jobs pane in Positron and VS Code by script, adopt familiar VS Code UI conventions, and open scripts or virtual Job Details documents when clicked
- (rstudio-pro#7862): Allow users in Positron and VS Code to dismiss completed jobs individually from the Workbench Jobs pane. Provide Dismiss Completed Jobs and Restore Completed Jobs actions to hide/unhide all completed jobs (not persistent across sessions).
- (rstudio-pro#8464): Add a preview setting in Workbench to launch jobs from Positron and VS Code with managed credentials (disabled by default)

> - (rstudio-pro#8148): Provide a new Audit Database and get_historical_session Workbench API endpoint to retrieve data about ended sessions

Reviewer 2: ```suggestion
- (rstudio-pro#8148): Provide a new Audit Database and `get_historical_session` Workbench API endpoint to retrieve data about ended sessions
```
Do we use backtics for API endpoints?

- (rstudio-pro#8124): Make Prometheus metrics generally available and remove Preview status

> - (rstudio-pro#8289): Remove pwb_session_duration from Prometheus Metrics and provide session start/end times via the new Workbench API endpoint get_historical_session

Reviewer 2: ```suggestion
- (rstudio-pro#8289): Remove `pwb_session_duration` from Prometheus Metrics and provide session start/end times via the new Workbench API endpoint `get_historical_session`
```
Same question here

- (rstudio-pro#6867): Enhance image selection in the Workbench launcher dialog to more clearly display and classify available images
- (rstudio-pro#8776): Add a switch in both Workbench homepages to toggle between the new and legacy versions and remember the user’s choice locally (see deprecation notice)
- (rstudio-pro#9009): Add the kapa.ai website widget to Posit Workbench documentation to answer questions about deploying, configuring, and using Workbench, Posit Connect, and Posit Package Manager.
- (rstudio-pro#7247): Add support for SLURM 24 and 25
- (rstudio-pro#8025): Improve diagnostic messages for communication errors between Workbench components
- (rstudio-pro#8502): Include version information for Positron Pro and Posit Workbench Code Server in diagnostic report output
- (vscode-server#164): Allow the location of VS Code session logs to be controlled with the `XDG_STATE_HOME` environment variable
- (vscode-server#202): Add GitHub Device Login URL as a default trusted domain for VS Code and Positron Pro sessions
- Improve Positron Admin and User Guide documentation

> - Add AWS Load Balanced Reference Architecture

Reviewer 3: Oh. I didn't know this went in the news file or I _would_ have submitted something for it. Thanks for covering it for me.

- (rstudio-pro#8462): Fixed an issue where VS Code/Positron Pro session browser state would not be loaded on sign in after being signed out due to inactivity
- (rstudio-pro#6867): Fixed an issue where image selection in the launcher UI was not behaving consistently
- (rstudio-pro#8690): Fixed an issue with locking users when using a license that has unlimited user count
- (rstudio-pro#7589) Fixed an issue where users were not gracefully returned to the homepage when clicking an invalid link
- (rstudio-pro#8080): Fixed an issue where the original CPU and Memory requests would not be remembered when resuming a session
- (rstudio-pro#8003): Fixed an issue where Positron user settings were not being copied to `user-data-dir`
- (rstudio-pro#8145): Fixed an issue where VS Code sessions were incorrectly labeled as `"rstudio-pro"` in the `pwb_sessions_launched_total` Prometheus metric
- (rstudio-pro#8563): Fixed an issue where the upload link in the Workbench diagnostics report was outdated
- (rstudio-pro#8580): Fixed an issue where sessions started in a Kubernetes cluster would fail to start if ".0000" was appended to the CPU request

> - (rstudio-pro#8929): Fixed an issue that could cause spurious login failures in Workbench
> - (rstudio-pro#8940): Fixed a deadlock issue causing `rstudio-server delete-user ...` to hang indefinitely

Author: ```suggestion
- (rstudio-pro#8929): Fixed an issue that could cause spurious login failures in Workbench
```

### Dependencies

- (rstudio-pro#8991): Nginx 1.29.1
- ([#15935](https://github.com/rstudio/rstudio/issues/15935)): Copilot Language Server 1.352.0
- ([#15933](https://github.com/rstudio/rstudio/issues/15933)): Electron 37.2.6
- ([#16062](https://github.com/rstudio/rstudio/issues/13924)): GWT 2.12.2

> - ([#15934](https://github.com/rstudio/rstudio/issues/15934)): Quarto 1.7.32

Reviewer 4: ```suggestion
- ([#15934](https://github.com/rstudio/rstudio/issues/15934)): Quarto 1.7.32
- (vscode-server#177): Code OSS for VSCode and Positron Pro sessions 1.102.0
- Node for VSCode and Positron Pro sessions 22.15.1
```

rstudio/vscode-server#177 isn't strictly about updating Code OSS but I believe it resolved the vulnerabilities in the issue and we don't have an issue that tracked updating code oss specifically AFAIK.
